instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/DefaultSpanStatusExtractor.java

```diff
@@ -14,11 +14,13 @@
   static final SpanStatusExtractor<Object, Object> INSTANCE = new DefaultSpanStatusExtractor<>();
 
   @Override
-  public StatusCode extract(
-      REQUEST request, @Nullable RESPONSE response, @Nullable Throwable error) {
+  public void extract(
+      SpanStatusBuilder spanStatusBuilder,
+      REQUEST request,
+      @Nullable RESPONSE response,
+      @Nullable Throwable error) {
     if (error != null) {
-      return StatusCode.ERROR;
+      spanStatusBuilder.setStatus(StatusCode.ERROR, error.toString());
```

Comment:
I'm unclear about [this part of the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status):

> When the status is set to `Error` by Instrumentation Libraries, the `Description` SHOULD be documented and predictable.

and whether an exception message is considered "predictable"

Reply:
I think that statement makes sense for error conditions as interpreted by the instrumentation, e.g. a failing status code or the like.  I can see where exceptions might be a bit murkier, but as long as the exception is being recorded as an event perhaps having it also be the description isn't as necessary (although I guess technically an exception event doesn't need to be _the_ reason that the span status is "Error" either).

I can certainly unwind the rest of the changes here so the exception message is not the description.  Would certainly make it easier to not have to deal with all of the exceptions that it broke.  😄

Reply:
I've reverted setting the error message as the description by default, that's something we can revisit.  This PR will instead only provide the `SpanStatusExtractor` that is capable of setting that description.